pkg/generate/app/pipeline.go

```diff
@@ -64,7 +64,7 @@ func (pb *pipelineBuilder) NewBuildPipeline(from string, resolvedMatch *Componen
 			return nil, fmt.Errorf("can't build %q: %v", from, err)
 		}
 		input = inputImage
-		if !input.AsImageStream {
+		if !input.AsImageStream && resolvedMatch.Value != "scratch" {
```

Comment:
well we do.  We mark the match as LocalOnly if we couldn't find it on dockhub (meaning we can't create an imagestream for it).  LocalOnly ultimately feeds into the value of AsImageStream.

The problem is that "scratch" is basically a localonly image (we can't/don't want to create an imagestream for it), but I don't want to spit out a warning for "scratch" because scratch is a special case of an image that doesn't exist as an imagestream, but doesn't require a warning.